crates/proof-of-sql/src/sql/ast/numerical_util.rs

```diff
@@ -121,6 +162,55 @@ pub(crate) fn add_subtract_columns<'a, S: Scalar>(
     res
 }
 
+// Add or subtract two owned columns together.
+pub(crate) fn add_subtract_owned_columns<S: Scalar>(
+    lhs: &OwnedColumn<S>,
+    rhs: &OwnedColumn<S>,
+    is_subtract: bool,
+) -> OwnedColumn<S> {
+    let lhs_len = lhs.len();
+    let rhs_len = rhs.len();
+    assert!(
+        lhs_len == rhs_len,
+        "lhs and rhs should have the same length"
+    );
+    let res_type = try_add_subtract_column_types(lhs.column_type(), rhs.column_type())
+        .expect("Operation is not supported");
+    // Check that lhs and rhs are numeric
+    assert!(lhs.column_type().is_numeric());
+    assert!(rhs.column_type().is_numeric());
+    let lhs_scale = lhs.column_type().scale().unwrap_or(0);
+    let rhs_scale = rhs.column_type().scale().unwrap_or(0);
+    let max_scale = lhs_scale.max(rhs_scale);
+    let lhs_upscale_factor =
+        scale_scalar(S::ONE, max_scale - lhs_scale).expect("Invalid scale factor");
+    let rhs_upscale_factor =
+        scale_scalar(S::ONE, max_scale - rhs_scale).expect("Invalid scale factor");
+    let res: Vec<S> = (0..lhs_len)
+        .map(|i| {
+            if is_subtract {
+                lhs.scalar_at(i).unwrap() * lhs_upscale_factor
+                    - rhs.scalar_at(i).unwrap() * rhs_upscale_factor
+            } else {
+                lhs.scalar_at(i).unwrap() * lhs_upscale_factor
+                    + rhs.scalar_at(i).unwrap() * rhs_upscale_factor
+            }
+        })
+        .collect();
+    OwnedColumn::Decimal75(
+        Precision::new(
+            res_type
+                .precision_value()
+                .expect("Precision of a valid column type should be valid"),
+        )
+        .expect("Precision of a valid column type should be valid"),
+        res_type
+            .scale()
+            .expect("Can't add or subtract non-numeric types"),
+        res,
+    )
+}
```

Comment:
Why is add and subtract together? Would it make more sense to have separate functions for add and subtract?

Reply:
@drcapybara They are together since add and subtract are essentially the same other than a sign.